### PR TITLE
Ana Luiza Bastos 

### PR DESCRIPTION
**Pull request by anabastos**
😸

### CODE REVIEW

File: exercices/anabastos/exercices/src/cipher/core.clj [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+(defn caesar-encrypt
+  "Encripta frase secreta por n"
+  [text n]
+  (apply str (mapv #(shift % n) (get-letters text))))
+
+(defn caesar-decrypt
+  "Decripta frase secreta por n"
+  [secret n]
+  (apply str (mapv #(shift % (* -1 n)) secret)))
```
Comment (isabelabel) [inline exercices/anabastos/exercices/src/cipher/core.clj]: `(* -1 n)` -> `(- n)`

File: exercices/anabastos/exercices/src/cipher/core.clj [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+(defn increase-frequency
+  "Função para aumentar a frequencia da palavra no hash"
+  [hash word]
+  (assoc hash word (inc (hash word 0))))
```
Comment (isabelabel) [inline exercices/anabastos/exercices/src/cipher/core.clj]: Take a look at `update` function. You could do something like:
```
(update hash word inc)
```

File: exercices/anabastos/exercices/src/cipher/core.clj [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+(defn build-frequency-map
+  "Controi hash de frequencia de todas as palavras da frase"
+  [word]
+  (reduce increase-frequency {} word))
+
+(defn count-letter
+  "Conta letras na frase"
+  [chr word]
+  (get (build-frequency-map word) chr 0))
```
Comment (isabelabel) [inline exercices/anabastos/exercices/src/cipher/core.clj]: 👏 Nice use of get with default value 0

File: exercices/anabastos/exercices/src/cipher/core.clj [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+  "Pega a diferença entre uma letra e a letra mais frequente no ingles(e, t, a)"
+  [word-most-frequent english-most-frequent]
+  (mod (- (to-int (char word-most-frequent)) (to-int (char english-most-frequent))) 26))
+
+(defn decrypt-by-key
+  "Decripta a partir das letras mais frequentes da frase"
+  [word-most-frequent test-english-key secret]
+  (let [key (get-key word-most-frequent test-english-key)]
+    (println "Key: " key " | Palavra:" word-most-frequent "| No ingles: " test-english-key)
+    (println (caesar-decrypt secret key))
```
Comment (isabelabel) [inline exercices/anabastos/exercices/src/cipher/core.clj]: Not necessary this `println`

**Comment by anabastos** [comment]
@isabelabel Thank you!!!

**Comment by marinalimeira** [comment]
unable to merge :(

**Pull request closed without merging**